File: python/descargar_pdf.py
import sys
from pathlib import Path

import requests

from config import BASE_URL, TOKEN


def obtener_comprobante_id() -> str:
    if len(sys.argv) < 2:
        raise RuntimeError(
            "Uso: python descargar_pdf.py ID_COMPROBANTE"
        )

    comprobante_id = sys.argv[1]

    if not comprobante_id.isdigit():
        raise RuntimeError(
            "El ID del comprobante debe ser un número entero."
        )

    return comprobante_id


def main() -> None:
    comprobante_id = obtener_comprobante_id()

    response = requests.get(
        f"{BASE_URL}/api/comprobantes/"
        f"{comprobante_id}/imprimir-a4",
        headers={
            "Authorization": f"Bearer {TOKEN}",
            "Accept": "application/pdf",
        },
        timeout=30,
    )

    if not response.ok:
        raise RuntimeError(
            f"Error {response.status_code}: {response.text}"
        )

    archivo = Path(f"comprobante-{comprobante_id}.pdf")
    temporal = archivo.with_suffix(".pdf.tmp")

    temporal.write_bytes(response.content)
    temporal.replace(archivo)

    print(f"PDF guardado en: {archivo}")


if __name__ == "__main__":
    main()

File: php/descargar_pdf.php
<?php

declare(strict_types=1);

require __DIR__ . '/config.php';

if ($argc < 2 || !ctype_digit($argv[1]) || (int) $argv[1] <= 0) {
    fwrite(STDERR, "Uso: php descargar_pdf.php ID_COMPROBANTE\n");
    exit(1);
}

$comprobanteId = $argv[1];
$response = apiRequest('GET', '/api/comprobantes/' . $comprobanteId . '/imprimir-a4', null, 'application/pdf');

if (!str_starts_with(strtolower($response['content_type']), 'application/pdf')) {
    throw new RuntimeException(
        'La API no devolvió un PDF válido. Content-Type recibido: ' . $response['content_type']
    );
}

$outputPath = __DIR__ . DIRECTORY_SEPARATOR . 'comprobante-' . $comprobanteId . '.pdf';
file_put_contents($outputPath, $response['body']);

echo 'PDF guardado en: ' . $outputPath . PHP_EOL;
